gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -116,139 +132,103 @@ public AbstractJobLauncher(Properties properties)
       taskExecutor.submit(task);
     }
 
-    LOG.info(String.format("Waiting for submitted tasks of job %s to complete...", jobId));
-    while (countDownLatch.getCount() > 0) {
-      LOG.info(String
-          .format("%d out of %d tasks of job %s are running", countDownLatch.getCount(), workUnits.size(), jobId));
-      countDownLatch.await(1, TimeUnit.MINUTES);
-    }
-    LOG.info(String.format("All tasks of job %s have completed", jobId));
-
     return tasks;
   }
 
+  /**
+   * This implementation simply sets a flag that indicate cancellation is requested and waits
+   * for the cancellation to be executed. It does not guarantee successful job cancellation.
+   *
+   * {@inheritDoc JobLauncher#cancelJob(JobListener)}
+   */
   @Override
-  @SuppressWarnings("unchecked")
-  public void launchJob(Properties jobProps, JobListener jobListener)
+  public void cancelJob(JobListener jobListener)
```

Comment:
I agree with the above points and what you are concerned about here, and it sounds like we need a thorough design on job state/lifecycle management before moving forward with any further changes around job cancellation handling. This is certainly beyond the scope of this pull request, which aims at providing a better default implementation around job cancellation than the previous one. Are you OK with the current default implementation?